### PR TITLE
Clarify CRLF behavior in multiline strings

### PR DESCRIPTION
**Pull request by brandonchinn178**
https://gitlab.haskell.org/ghc/ghc/-/issues/25375

### CODE REVIEW

**Comment by VitWW** [comment]
1) It is better update the title of Proposal to something like "Amend multiline strings to clarify CRLF behavior"
2) Maybe it is better not to change all "newlines" to "line terminators", but to add additional first step:
```
#. Convert CRLF "\r\n" line terminators to LF "\n" line terminators.
```

**Comment by sgraf812** [comment]
I had a look at the [Haskell report](https://www.haskell.org/onlinereport/haskell2010/haskellch2.html#x7-200002.6). It defines

```
newline 	→ 	return linefeed | return | linefeed | formfeed
return 	→ 	a carriage return
linefeed 	→ 	a line feed
formfeed 	→ 	a form feed
```

See https://stackoverflow.com/questions/3091524/what-are-carriage-return-linefeed-and-form-feed for what's what.

In light of the specification, I do not see how what this PR improves, because it does not define what a "line terminator" is.

On the bright side, it sounds like https://gitlab.haskell.org/ghc/ghc/-/issues/25375 is not a bug in the specification at all! No need for a GHC proposal. Would you agree, @brandonchinn178?

**Comment by brandonchinn178** [comment]
@sgraf812, yeah that sounds good, how's that?

File: proposals/0569-multiline-strings.rst [hunk 1]
```diff
@@ -117,8 +121,6 @@ A working prototype is available at `brandonchinn178/string-syntax <https://gith
 
    #. Join the string back with ``\n`` delimiters
 
-      * Use ``\n`` regardless of the line terminators being used in the file. This matches the behavior of ``unlines``.
-
```
Comment (sgraf812) [inline proposals/0569-multiline-strings.rst:121]: I would prefer for the bullet to stay here, where it makes perfect sense. Perhaps replace "line terminators" by "lexical ``newline`` characters"

File: proposals/0569-multiline-strings.rst [hunk 2]
```diff
@@ -98,6 +98,10 @@ A working prototype is available at `brandonchinn178/string-syntax <https://gith
       * See `Section 2.6 <https://www.haskell.org/onlinereport/haskell2010/haskellch2.html#x7-200002.6>`_ of the Haskell 2010 Report
       * See the example in *Section 3.3 String gaps*
 
+   #. Convert all lexical ``newline`` characters (as defined in the `Haskell report <https://www.haskell.org/onlinereport/haskell2010/haskellch2.html#x7-160002.2>`_) to ``\n``
```
Comment (sgraf812) [inline proposals/0569-multiline-strings.rst:101]: I think it's a bit redundant to first *convert* all lexical `newline` characters to `\n`, only to then *split* the string at all such `\n`.

Furthermore, you say `\n` here, but "Split the string by newlines" later. If you going to split by (lexical) newlines anyway, the conversion step seems unnecessary. If you do not mean lexical newline, but rather only `\n`, then perhaps say so.

That said, I would favour the more minimal change that just clarifies the "Split the string by newlines" step and gets rid of this additional conversion step entirely (which seems like an implementation detail at best).

Comment (brandonchinn178) [reply proposals/0569-multiline-strings.rst]: Sure, udpated

**Comment by sgraf812** [comment]
LGTM! Do you want to submit this proposal to the committee? I would be willing to shepherd.

**Comment by brandonchinn178** [comment]
Yes please, thank you!

**Comment by adamgundry** [comment]
Thanks for working on this @brandonchinn178, and thanks @sgraf812 for volunteering to shepherd!

**Comment by sgraf812** [comment]
I recommend that we accept this proposal. It is merely a small clarifying amendment that addresses a bug in GHC caused by misinterpretation of the proposal text pre amendment.

**Comment by erikd** [comment]
Accept.

File: proposals/0569-multiline-strings.rst [hunk 3]
```diff
@@ -117,7 +117,7 @@ A working prototype is available at `brandonchinn178/string-syntax <https://gith
 
    #. Join the string back with ``\n`` delimiters
 
-      * Use ``\n`` regardless of the line terminators being used in the file. This matches the behavior of ``unlines``.
+      * Use ``\n`` regardless of the lexical ``newline``s present in the file. This matches the behavior of ``unlines``.
```
Comment (JakobBruenker) [inline proposals/0569-multiline-strings.rst:120]: it looks like the `s` immediately following the inline code messes up the formatting. This seems to fix it.
```suggestion
      * Use ``\n`` regardless of the lexical ``newline``\ s present in the file. This matches the behavior of ``unlines``.
```

Comment (akhra) [reply proposals/0569-multiline-strings.rst:120]: does this work:
```
``newline``(s)
```

Comment (JakobBruenker) [reply proposals/0569-multiline-strings.rst:120]: I tried it in a gist, but it looks like that is also formatted incorrectly

**Comment by sgraf812** [comment]
The committee has voted to accept this amendment proposal.

**Comment by brandonchinn178** [comment]
@sgraf812 Thanks! Shall we merge this, given the MR has merged?

**Comment by adamgundry** [comment]
Thanks for your work on this @brandonchinn178!

@sgraf812 I've merged, but in general when acting as shepherd you are also welcome to merge accepted proposals yourself. (For full proposals there are a few more steps, [documented here](https://github.com/ghc-proposals/ghc-proposals/blob/master/acceptance.rst).)

**Pull request merged**